docs/developer-experience/readme.md

```diff
@@ -0,0 +1,231 @@
+
+The long answer is because:
+
+- sometimes the decorators *are* mandatory, and it is easier to use them all the time.
+- the decorators, even when optional, mark a function as belonging to a step or a pipeline, and help differentiate the code which is a step from normal functions.
+- the decorators may provide helpful intellisense help[^4]
+
+As an example, imagine you want to make a "guess" service to receive the value head or tail and then check if this guess is valid or not. Using bauta.js you might end with something similar to this:
+
+```javascript
+
+const guessPipeline = pipe(
+  obtainCoin,
+  getGuessResult,
+  iif(isGoodGuess, youHaveWon, youHaveLost)
+);
+
+```
+Using this pattern, every function has a single responsability and at the pipeline level, the code that glues the logic of the function can be understood easily just looking at the pipeline. 
+
+Furthermore, you can create a pipeline and pipe it into another pipeline, which allows patterns for composing and reusing pipelines in more than one operation or services.
+
+## Developer experience comparison 
+
+Let's assume that you want to implement an API REST service. Regardless of its functional domain or its technical requirements, there are a few things that the service will need to do.
+
+In the next diagram in the left side we convey these responsabilities if you do not use bauta.js.
+
+![bauta.js conceptual comparison](./service-conceptual-hlv.png)
+
+We are not saying that your application code has to implement all those responsabilities from scrath. In fact it is almost sure that you will use libraries for that, but the point is that you will have those libraries as dependencies from your project and you will have to maintain those dependencies and the code that integrates them in your application. 
+
+In comparison you can see that on the right side, with bauta.js, you have those responsabilities hidden inside bauta.js. You can use them as easily with your custom solution but your service has less boilerplate due to those common technical requirements and you can focus on your business code. 
+
+### Small note about the diagram and fastify
+
+While the diagram represents 100% a case using the express framework, it is not totally exact for a case using fastify. The reason is because fastify offers its own high performant schema validator and body parser. Thus, for the diagram in the right to be totally correct the body parser box should be inside fastify box and there should be up two schema validations boxes: one for bauta.js and the other for fastify that validates as well when parsing the request. 
+
+Another way of seeing this is that when using fastify, bauta.js is used as plugin. In any case the message is the same: your application does not have to bother with schema validation or other technical details because is being done by the abstractions provided by bauta.js together with the chosen server framekwork.
```

Comment:
```suggestion
While the diagram represents 100% a case using the Express.js framework, it is not precisely the same if you use fastify. This is because fastify offers its own high-performance schema validator and body parser. Thus, for the diagram at the right to be correct, the body parser box should be inside the fastify box, and there should be up to two schema validations boxes: one for bauta.js and the other for fastify that validates as well when parsing the request. 

Another way of seeing this is that when using fastify, bauta.js is used as a plugin. The message is the same: your application does not bother with schema validation or other technical details because it is done by the abstractions provided by bauta.js and the chosen server framework.
```